### PR TITLE
Move gas/brake pedal event logic to controlsd

### PR DESCRIPTION
**Pull request by sshane**
Makes it so we only have to read the disengage on gas param once, and the car interfaces probably don't need to know about the toggle

### CODE REVIEW

**Comment by sshane** [comment]
I believe the diffs are just order based, the final alert is whatever is added first if the priority and start frame is the same

File: selfdrive/controls/controlsd.py [hunk 1]
```diff
@@ -192,6 +193,10 @@ def update_events(self, CS):
       self.events.add(EventName.controlsInitializing)
       return
 
+    # Disable on rising edge of gas or brake. Also disable on brake when speed > 0
+    if (CS.gasPressed and self.active) or (CS.brakePressed and not CS.standstill):
```
Comment (adeebshihadeh) [inline selfdrive/controls/controlsd.py:197]: this isn't the same as before. now, we don't disengage on the rising edge of brake

File: selfdrive/controls/controlsd.py [hunk 2]
```diff
@@ -192,6 +194,11 @@ def update_events(self, CS):
       self.events.add(EventName.controlsInitializing)
       return
 
+    # Disable on rising edge of gas or brake. Also disable on brake when speed > 0
+    if (CS.gasPressed and self.active) or (CS.brakePressed and (not self.brake_pressed_prev or not CS.standstill)):
+      self.events.add(EventName.pedalPressed)
```
Comment (adeebshihadeh) [inline selfdrive/controls/controlsd.py:199]: let's save the previous car state so we can do the same check. it's unclear that `CS.gasPressed and self.active` enforces `Disable on rising edge of gas`

File: selfdrive/controls/controlsd.py [hunk 3]
```diff
@@ -733,6 +740,7 @@ def step(self):
     self.prof.checkpoint("Sent")
 
     self.update_button_timers(CS.buttonEvents)
+    self.CS_prev = CS
```
Comment (adeebshihadeh) [inline selfdrive/controls/controlsd.py:743]:
```suggestion
    self.CS_prev = CS.as_reader()
```

Comment (sshane) [reply selfdrive/controls/controlsd.py:743]: Already a reader, from CarInterface: `self.CS.out = ret.as_reader()`

**Comment by sshane** [comment]
One of the specific routes with differences:

Screenshot from 2022-03-15 19-46-59

Looks okay to me. The active event is the first added event, so now we do the `pedalPressed` event check first and on rising edge of gas it's active for one frame.

**Pull request merged**